impl/src/java/org/sakaiproject/poll/service/impl/PollVoteManagerImpl.java

```diff
@@ -137,11 +140,16 @@ public boolean userHasVoted(Long pollid, String userID) {
 		Search search = new Search();
 		search.addRestriction(new Restriction("userId",userID));
 		search.addRestriction(new Restriction("pollId",pollid));
-		List<Vote> votes = dao.findBySearch(Vote.class, search);		
-		if (votes.size() > 0)
-			return true;
-		else
+		try {
+			List<Vote> votes = dao.findBySearch(Vote.class, search);
+			if (votes.size() > 0)
+				return true;
+			else
+				return false;
+		}catch(Exception e) {
```

Comment:
It's bad form todo `catch (Exception e)` with something like this, it means that if you have something like a `NullPointerException` it's really hard to track down.

We should really be expecting Exceptions like this in the normal code path.